features/FEATURE_BLE/targets/TARGET_CORDIO/stack_adaptation/hci_tr.c

```diff
@@ -66,11 +66,12 @@ void hciTrSendAclData(void *pContext, uint8_t *pData)
   HCI_PDUMP_TX_ACL(len, pData);
 
   /* transmit ACL header and data */
-  if (hciDrvWrite(HCI_ACL_TYPE, len, pData) == len)
+  if (hciDrvWrite(HCI_ACL_TYPE, len, pData) != len)
   {
-    /* free buffer */
-    hciCoreTxAclComplete(pContext, pData);
+    /* transport failure */
```

Comment:
Nope, that's going to break the driver on targets that don't use zero copy (All of them except the new Cordio LL). This can be solved with a compile time configuration; if the target support zero copy then we do not release the data, otherwise we release the pointer. No zero copy should be the default mode.

Reply:
config option added